# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 6e9a89ead3a5
# Full Hash: 6e9a89ead3a52864c48a4e5481e9a35e4aad5e98
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r dd00e2da3a0f -r 6e9a89ead3a5 dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Wed Jun 10 22:21:02 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Wed Jun 10 22:21:04 2020 +0000
@@ -100,6 +100,7 @@
 
 namespace layers {
 class CompositableHost;
+class SurfaceDescriptor;
 }
 
 namespace webgl {
@@ -484,9 +485,10 @@
   bool PresentInto(gl::SwapChain& swapChain);
 
  public:
-  void Present();
+  void Present(WebGLFramebuffer*, layers::TextureType);
   RefPtr<gfx::DataSourceSurface> GetFrontBufferSnapshot();
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(layers::TextureType);
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebuffer*,
+                                                  layers::TextureType);
 
   void RunContextLossTimer();
   void CheckForContextLoss();